core/src/mindustry/net/ArcNetProvider.java

```diff
@@ -189,8 +192,27 @@ public void sendClient(Object object, boolean reliable){
     @Override
     public void pingHost(String address, int port, Cons<Host> valid, Cons<Exception> invalid){
         try{
-            DatagramSocket socket = new DatagramSocket();
+            var host = pingHostImpl(address, port);
+            Core.app.post(() -> valid.get(host));
+        }catch(IOException e){
+            if(port == Vars.port){
+                for (var record : ArcDns.getSrvRecords("_mindustry._tcp." + address)) {
+                    try{
+                        var host = pingHostImpl(record.target, record.port);
+                        Core.app.post(() -> valid.get(host));
+                        return;
+                    }catch(IOException ignored){
+                    }
+                }
+            }
+            Core.app.post(() -> invalid.get(e));
+        }
+    }
+
+    private Host pingHostImpl(String address, int port) throws IOException {
```

Comment:
unnecessary space before {

Reply:
> unnecessary space before {

Anuke why do you not just...
```suggestion
    private Host pingHostImpl(String address, int port) throws IOException{
```

Reply:
g